### ISSUE TITLE
ArrayIndexOutOfBoundsException (@LegendRenderer:renderLegend:239)

### ISSUE THREAD

**Issue by ste71m** [PhilJay/MPAndroidChart#952]
I'm getting this exception from some user
ArrayIndexOutOfBoundsException (@LegendRenderer:renderLegend:239)
Sorry i don't have the full stack trace or other infos
Could you check getCalculatedLineSizes size in that line to avoid to throw exceptions?

**Comment by aint** [comment]
I have almost the same problem.
```java
java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
            at com.github.mikephil.charting.renderer.LegendRenderer.renderLegend(LegendRenderer.java:233)
            at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:256)
```
I noticed that app crashes when ```LegendPosition.BELOW_CHART_CENTER``` set. With ```RIGHT_OF_CHART_INSIDE``` all works well.

**Comment by ste71m** [comment]
got a stack tarce

java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
at com.github.mikephil.charting.renderer.LegendRenderer.renderLegend(SourceFile:239)
at com.github.mikephil.charting.charts.PieChart.onDraw(SourceFile:114)

**Comment by PhilJay** [comment]
Are you adding data dynamically?

**Comment by ste71m** [comment]
No, data don't change once chart  (pie chart) has been created
And issue appeared in last versions, before it was working fine.
As aint said, changing LegendPosition.BELOW_CHART_CENTER to  LegendPosition.RIGHT_OF_CHART_INSIDE seems to fix

**Comment by PhilJay** [comment]
Its not clear to me how this issue can arise, but I have added bounds checks for the arrays that should prevent it.

**Comment by ste71m** [comment]
Thanks. I tried but could not replicate the issue so have no idea too which condition could cause exception.

**Comment by PhilJay** [comment]
I suppose the issue is now gone because you just fixed it by calling `notifyDataSetChanged()` beforehand!

**Comment by aint** [comment]
In my case calling ```notifyDataSetChanged()``` have no effect.
```java
mChart.setData(...);
Legend legend = mChart.getLegend();
legend.setCustom(new ArrayList<>(colors), new ArrayList<>(labels));
legend.setPosition(Legend.LegendPosition.BELOW_CHART_CENTER);
legend.setWordWrapEnabled(true);
mChart.invalidate();
```

```java
java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
            at com.github.mikephil.charting.renderer.LegendRenderer.renderLegend(LegendRenderer.java:233)
            at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:256)
```

**Comment by kitwood** [comment]
FWIW, I just ran into this issue. I didn't have any troubles on my Nexus 5, but I got the ArrayIndexOutOfBoundsException when running on my Moto G (3rd gen). It looks like there's some rounding issue when it calculates the line heights and I suppose there's some variations on platform. I found I could fix it by calling `pieChart.getLegend().setTextSize(11f)`. If I set the legend text size to 10f, I'd crash, but bumping it to 11 (or larger), and I was fine.
